Move lcd tests into gaia

alessio opened the pull request:
This is to remove lcd -> gaia package dep.


- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry: `clog add [section] [stanza] [message]`
- [ ] rereviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

alexanderbez commented:
So we're going to have the LCD tests live in the to-be gaia repo and CLI tests live the SDK repo?

alessio commented:
@alexanderbez I am afraid that we're going to have both LCD and CLI tests in the to-be gaia repo only. They all are gaia-specific at this moment. I wouldn't put too much effort into bettering lcd tests as the mid-term plan is to drop them in favor of the contract tests @sabau is working on.

alessio commented:
I'll fix that @jackzampolin as discussed. Thanks!

alessio commented:
Removed `lcd`'s private `cdc` variable as it was used by the tests that now live elsewhere. I've also removed a unused function in lcd's `root.go`

Should be ready to go @jackzampolin @alexanderbez
